processing/src/main/java/io/druid/query/filter/IntervalDimFilter.java

```diff
@@ -0,0 +1,178 @@
+import org.joda.time.Interval;
+
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class IntervalDimFilter implements DimFilter
+{
+  private final List<Interval> intervals;
+  private final String dimension;
+  private final ExtractionFn extractionFn;
+  private final List<DimFilter> boundDimFilters;
+
+  @JsonCreator
+  public IntervalDimFilter(
+      @JsonProperty("dimension") String dimension,
+      @JsonProperty("intervals") List<Interval> intervals,
+      @JsonProperty("extractionFn") ExtractionFn extractionFn
+  )
+  {
+    Preconditions.checkNotNull(dimension, "dimension can not be null");
+    Preconditions.checkNotNull(intervals, "intervals can not be null");
+    Preconditions.checkArgument(intervals.size() > 0, "must specify at least one interval");
+    this.dimension = dimension;
+    this.intervals = Collections.unmodifiableList(JodaUtils.condenseIntervals(intervals));
+    this.extractionFn = extractionFn;
+    this.boundDimFilters = makeBoundDimFilters();
+  }
+
+  @JsonProperty
+  public String getDimension()
+  {
+    return dimension;
+  }
+
+  @JsonProperty
+  public List<Interval> getIntervals()
+  {
+    return intervals;
+  }
+
+  @JsonProperty
+  public ExtractionFn getExtractionFn()
+  {
+    return extractionFn;
+  }
+
+  @Override
+  public byte[] getCacheKey()
+  {
+    byte[] dimensionBytes = StringUtils.toUtf8(dimension);
+
+    final byte[][] intervalsBytes = new byte[intervals.size()][];
+    int intervalsBytesSize = 0;
+    int index = 0;
+    for (Interval interval : intervals) {
+      intervalsBytes[index] = StringUtils.toUtf8(interval.toString());
+      intervalsBytesSize += intervalsBytes[index].length + 1;
+      ++index;
+    }
+    byte[] extractionFnBytes = extractionFn == null ? new byte[0] : extractionFn.getCacheKey();
+
+    ByteBuffer filterCacheKey = ByteBuffer.allocate(3
+                                                    + dimensionBytes.length
+                                                    + intervalsBytesSize
+                                                    + extractionFnBytes.length)
+                                          .put(DimFilterUtils.INTERVAL_CACHE_ID)
+                                          .put(dimensionBytes)
+                                          .put(DimFilterUtils.STRING_SEPARATOR)
+                                          .put(extractionFnBytes)
+                                          .put(DimFilterUtils.STRING_SEPARATOR);
+    for (byte[] bytes : intervalsBytes) {
+      filterCacheKey.put(bytes)
+                    .put((byte) 0xFF);
+    }
+    return filterCacheKey.array();
+  }
+
+  @Override
+  public DimFilter optimize()
+  {
+    return this;
+  }
+
+  @Override
+  public Filter toFilter()
+  {
+    return new OrDimFilter(boundDimFilters).toFilter();
+  }
+
+  @Override
+  public RangeSet<String> getDimensionRangeSet(String dimension)
```

Comment:
This method could be supported. You could do it without too much work by doing `new OrDimFilter(boundDimFilters).getDimensionRangeSet(dimension)`